rust/kernel/fs.rs

```diff
@@ -49,6 +49,14 @@ pub mod ro {
 
         /// Reads the contents of the inode into the given folio.
         fn read_folio(inode: &INode<Self>, folio: crate::folio::LockedFolio<'_>) -> Result;
+
+        /// Reads an xattr.
+        ///
+        /// Returns the number of bytes written to `outbuf`. If it is too small, returns the number
+        /// of bytes needs to hold the attribute.
```

Comment:
I am a bit skeptical about returning `Ok` when outbuf is too small. What about `Result<Either<usize, usize>>` (not sure if we have `Either` already in the kernel, but Alice also needs it for binder)?

Reply:
This is the way the C API works: a negative number indicates an error, non-negative numbers indicate the length how much is written or the total size, whichever is greater.

Either is already merged upstream. But what would this accomplish?

The implementation of read_xattr for tarfs ends as follows:

```rust
        if !outbuf.is_empty() {
            outbuf[0] = b'y';
        }

        Ok(1)
```

Reply:
it was just a little weird to me